src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,161 @@
+            resultSet.next();
+            user = getUserFromResultSet(resultSet);
+            user = getRolesForUser(user);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get user from DataBase");
+        }
+        return Optional.of(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        String query = "SELECT * FROM users";
+        List<User> users = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                users.add(getUserFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get users in DataBase");
+        }
+        return users;
+    }
+
+    @Override
+    public User update(User element) {
+        String query = "UPDATE users SET name = ?, login = ?, password = ?  WHERE user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, element.getName());
+            statement.setString(2, element.getLogin());
+            statement.setString(3, element.getPassword());
+            statement.setLong(4, element.getId());
+            statement.execute();
+            return element;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update user in DataBase");
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM users WHERE user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.execute();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete user in DataBase");
+        }
+    }
+
+    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
+        User user = new User();
+        Long userId = resultSet.getLong("user_id");
+        String name = resultSet.getString("name");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
+        user.setId(userId);
+        user.setName(name);
+        user.setLogin(login);
+        user.setPassword(password);
+        return user;
+    }
+
+    private User getRolesForUser(User user) throws SQLException {
+        String query = "SELECT * FROM users INNER JOIN user_roles "
+                + "ON users.user_id = user_roles.user_id "
+                + "INNER JOIN roles on user_roles.role_id = roles.role_id "
+                + "WHERE users.user_id = ?";
```

Comment:
do you need to join users table to get user's roles?

Reply:
done